src/main/java/org/zalando/nakadi/service/SchemaService.java

```diff
@@ -48,18 +47,16 @@ public SchemaService(final SchemaRepository schemaRepository,
     public Result<EventTypeSchema> getSchemaVersion(final String name, final String version) {
         final Matcher versionMatcher = VERSION_PATTERN.matcher(version);
         if (!versionMatcher.matches()) {
-            return Result.problem(Problem.valueOf(Response.Status.NOT_FOUND));
+            return Result.problem(Problem.valueOf(Response.Status.BAD_REQUEST, "Invalid version number"));
         }
 
         try {
             final EventTypeSchema schema = schemaRepository.getSchemaVersion(name, version);
             return Result.ok(schema);
         } catch (final NoSuchSchemaException e) {
             LOG.debug("Could not find EventTypeSchema version: {} for EventType: {}", version, name);
-            return Result.problem(e.asProblem());
-        } catch (final InternalNakadiException e) {
-            LOG.error("Problem loading event type schema version " + version + " for EventType " + name, e);
-            return Result.problem(e.asProblem());
+            return Result.problem(Problem.valueOf(Response.Status.NOT_FOUND,
```

Comment:
Done in https://github.com/zalando/nakadi/commit/25ef9b2d8c714d0c6d384f5b42e52ef5a05904e0